### PR TITLE
Refresh the Azure.ResourceManager.Communication up to '2021-10-01-preview'

### LINKED ISSUE #28877: Refresh the Azure.ResourceManager.Communication up to '2021-10-01-preview'
The spec is still in https://github.com/Azure/azure-rest-api-specs-pr/tree/RPSaaSMaster/specification/communication/resource-manager/Microsoft.Communication/preview/2021-10-01-preview, will be merged in May 23.

### PR DESCRIPTION
**Pull request by HarveyLink**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/communication/Azure.ResourceManager.Communication/src/Generated/EmailServiceResourceData.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        internal EmailServiceResourceData(ResourceIdentifier id, string name, ResourceType resourceType, SystemData systemData, IDictionary<string, string> tags, AzureLocation location, EmailServicesProvisioningState? provisioningState, string dataLocation) : base(id, name, resourceType, systemData, tags, location)
+        {
+            ProvisioningState = provisioningState;
+            DataLocation = dataLocation;
+        }
+
+        /// <summary> Provisioning state of the resource. </summary>
+        public EmailServicesProvisioningState? ProvisioningState { get; }
+        /// <summary> The location where the email service stores its data at rest. </summary>
+        public string DataLocation { get; set; }
```
[the remark this answers is not in the document]
Comment (HarveyLink) [reply sdk/communication/Azure.ResourceManager.Communication/src/Generated/EmailServiceResourceData.cs:42]: Value goes to "UnitedStates", not `AzureLocation`.

File: sdk/communication/Azure.ResourceManager.Communication/src/Generated/DomainResourceData.cs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            VerificationStates = verificationStates;
+            VerificationRecords = verificationRecords;
+            ValidSenderUsernames = validSenderUsernames;
+            UserEngagementTracking = userEngagementTracking;
+        }
+
+        /// <summary> Provisioning state of the resource. </summary>
+        public DomainsProvisioningState? ProvisioningState { get; }
+        /// <summary> The location where the Domains resource data is stored at rest. </summary>
+        public string DataLocation { get; }
```
[the remark this answers is not in the document]
Comment (HarveyLink) [reply sdk/communication/Azure.ResourceManager.Communication/src/Generated/DomainResourceData.cs:57]: Value goes to "UnitedStates", not `AzureLocation`.

File: sdk/communication/Azure.ResourceManager.Communication/src/Generated/DomainResource.cs [hunk 3]
```diff
@@ -0,0 +1,551 @@
...
+
+namespace Azure.ResourceManager.Communication
+{
+    /// <summary>
+    /// A Class representing a DomainResource along with the instance operations that can be performed on it.
+    /// If you have a <see cref="ResourceIdentifier" /> you can construct a <see cref="DomainResource" />
+    /// from an instance of <see cref="ArmClient" /> using the GetDomainResource method.
+    /// Otherwise you can get one from its parent resource <see cref="EmailServiceResource" /> using the GetDomainResource method.
+    /// </summary>
+    public partial class DomainResource : ArmResource
```
[the remark this answers is not in the document]
Comment (HarveyLink) [reply sdk/communication/Azure.ResourceManager.Communication/src/Generated/DomainResource.cs]: Renamed with CommunicationDomainResource

File: sdk/communication/Azure.ResourceManager.Communication/src/Generated/DomainResource.cs [hunk 4]
```diff
@@ -0,0 +1,551 @@
...
+        /// Operation Id: Domains_InitiateVerification
+        /// </summary>
+        /// <param name="waitUntil"> <see cref="WaitUntil.Completed"/> if the method should wait to return until the long-running operation has completed on the service; <see cref="WaitUntil.Started"/> if it should return after starting the operation. For more information on long-running operations, please see <see href="https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/LongRunningOperations.md"> Azure.Core Long-Running Operation samples</see>. </param>
+        /// <param name="verificationParameter"> Type of verification to be initiated. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="verificationParameter"/> is null. </exception>
+        public virtual async Task<ArmOperation> InitiateVerificationAsync(WaitUntil waitUntil, VerificationParameter verificationParameter, CancellationToken cancellationToken = default)
```
[the remark this answers is not in the document]
Comment (HarveyLink) [reply sdk/communication/Azure.ResourceManager.Communication/src/Generated/DomainResource.cs]: Updated.

File: sdk/communication/Azure.ResourceManager.Communication/src/Generated/Extensions/CommunicationExtensions.cs [hunk 5]
```diff
@@ -31,53 +31,111 @@ private static SubscriptionResourceExtensionClient GetExtensionClient(Subscripti
...
         /// Request Path: /subscriptions/{subscriptionId}/providers/Microsoft.Communication/checkNameAvailability
-        /// Operation Id: CommunicationService_CheckNameAvailability
+        /// Operation Id: CommunicationServices_CheckNameAvailability
         /// </summary>
         /// <param name="subscriptionResource"> The <see cref="SubscriptionResource" /> instance the method will execute against. </param>
         /// <param name="content"> Parameters supplied to the operation. </param>
         /// <param name="cancellationToken"> The cancellation token to use. </param>
-        public static async Task<Response<NameAvailability>> CheckCommunicationNameAvailabilityAsync(this SubscriptionResource subscriptionResource, NameAvailabilityContent content = null, CancellationToken cancellationToken = default)
+        /// <exception cref="ArgumentNullException"> <paramref name="content"/> is null. </exception>
+        public static async Task<Response<CheckNameAvailabilityResponse>> CheckNameAvailabilityCommunicationServiceAsync(this SubscriptionResource subscriptionResource, NameAvailabilityContent content, CancellationToken cancellationToken = default)
```
[the remark this answers is not in the document]
Comment (HarveyLink) [reply sdk/communication/Azure.ResourceManager.Communication/src/Generated/Extensions/CommunicationExtensions.cs]: Fixed

File: sdk/communication/Azure.ResourceManager.Communication/src/Generated/Models/CheckNameAvailabilityRequest.cs [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+namespace Azure.ResourceManager.Communication.Models
+{
+    /// <summary> The check availability request body. </summary>
+    public partial class CheckNameAvailabilityRequest
```
[the remark this answers is not in the document]
Comment (HarveyLink) [reply sdk/communication/Azure.ResourceManager.Communication/src/Generated/Models/CheckNameAvailabilityRequest.cs]: Renamed with CheckNameAvailabilityRequestBody

File: sdk/communication/Azure.ResourceManager.Communication/src/Generated/Models/DomainPropertiesVerificationRecords.cs [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+            SPF = spf;
+            Dkim = dkim;
+            Dkim2 = dkim2;
+            Dmarc = dmarc;
+        }
+
+        /// <summary> A class that represents a VerificationStatus record. </summary>
+        public DnsRecord Domain { get; }
+        /// <summary> A class that represents a VerificationStatus record. </summary>
+        public DnsRecord SPF { get; }
```
[the remark this answers is not in the document]
Comment (HarveyLink) [reply sdk/communication/Azure.ResourceManager.Communication/src/Generated/Models/DomainPropertiesVerificationRecords.cs]: Updated

File: sdk/communication/Azure.ResourceManager.Communication/src/Generated/Models/DomainPropertiesVerificationStates.cs [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+            SPF = spf;
+            Dkim = dkim;
+            Dkim2 = dkim2;
+            Dmarc = dmarc;
+        }
+
+        /// <summary> A class that represents a VerificationStatus record. </summary>
+        public VerificationStatusRecord Domain { get; }
+        /// <summary> A class that represents a VerificationStatus record. </summary>
+        public VerificationStatusRecord SPF { get; }
```
[the remark this answers is not in the document]
Comment (HarveyLink) [reply sdk/communication/Azure.ResourceManager.Communication/src/Generated/Models/DomainPropertiesVerificationStates.cs]: Updated

**Review by Yao725 — APPROVED** [review]
LGTM in general.

**Pull request merged**